src/modules/esl-utils/async/throttle.ts

```diff
@@ -26,21 +26,21 @@ export function throttle<F extends AnyToAnyFnSignature>(fn: F, threshold = 250,
 
   function throttledSubject(...args: any[]): void {
     const now = Date.now();
+    const lastThreshold = last + threshold;
 
-    if (!last || now >= last + threshold) {
-      last = now;
-      fn.apply(this, args);
-    }
+    deferred = createDeferred();
```

Comment:
The behavior of the result promise changed.
First of all, it's changed incorrectly: as for right now, it will produce promises that never be resolved (memory leak).
Second thing, not sure about the deferred and result promise in general for `throttle`.
We need to have a logical principal for result promise resolution, or we need to consider getting rid of that feature at all for `throttle`